App/Infrastructure/Laravel/Contracts/EventDispatcher.php

```diff
@@ -0,0 +1,10 @@
+<?php
+
+namespace App\Infrastructure\Laravel\Contracts;
+
+use Illuminate\Contracts\Events\Dispatcher;
+
+interface EventDispatcher extends Dispatcher
+{
+    public function flushAll(array $events);
+}
```

Comment:
Would it make sense for this to be decoupled from Illuminate's contract using some domain abstraction? 
I'm used to avoid as much as possible making the application layer depend on the infrastructure so I'm curious about the rationale/rule here :)

Reply:
The EventDispatcher contract and the Dispatcher class implementation for the events are extensions of Laravel's eventing system. 

It does not support flushing multiple events at once so I added it manually.

I plan to create a PR to Laravel to move this logic to the Illuminate namespace (if it gets accepted).

Maybe I will move this elsewhere for the time being, thanks for the suggestion.

Reply:
I moved the EventDispatcher contract to the `App/Application/Events` namespace and implemented a FakeEventDispatcher contract in the `Laravel/Events` namespace. 

Let me know what you think about this change.